driver/src/main/java/oracle/nosql/driver/values/JsonUtils.java

```diff
@@ -758,4 +758,38 @@ public static String convertBytesToHex(byte[] byteArray) {
         }
         return new String(hexValue);
     }
+
+    /**
+     * Validates input is a valid jsonObject. Throws IllegalArgumentException if
+     * not.
+     */
+    public static void validateJsonObject(String jsonInput) {
+        try (JsonParser jp = createParserWithOptions(jsonInput, null)) {
+            JsonToken token = jp.nextToken();
+            if (token == null || !JsonToken.START_OBJECT.equals(token)) {
+                throw new IllegalArgumentException("Not a JSON Object");
+            }
+            int s = 1;
+            while (!jp.isClosed()) {
+                token = jp.nextToken();
+                if (token != null && JsonToken.START_OBJECT.equals(token)) {
+                    if (s == 0) {
+                        throw new IllegalArgumentException("Multiple JSON " +
```

Comment:
Is this multiple top-level objects or nested objects? If the latter why the restriction?

Reply:
It's the latter, it's not allowed because it is not a valid json object.